collector/generic_collector.go

```diff
@@ -45,8 +57,20 @@ func NewCollector(collectorName string, configfile string) (*GenericCollector, e
 
 	//TODO : Add checks for validity of config file (eg : Accurate JSON fields)
 
+	minPollFrequency := configInJSON.MetricsConfig[0].PollingFrequency
+	regexprs := make([]*regexp.Regexp, len(configInJSON.MetricsConfig))
+
+	for ind, metricConfig := range configInJSON.MetricsConfig {
+		if metricConfig.PollingFrequency < minPollFrequency {
+			minPollFrequency = metricConfig.PollingFrequency
+		}
+
+		//regexprs[ind] is set to nil if metricConfig.Regex cannot be parsed
+		regexprs[ind], _ = regexp.Compile(metricConfig.Regex)
```

Comment:
I see the point It might be  unclear to the user how only a few metrics have been collected. Thanks @rjnagal @vmarmol . Will fix it.